Helpers/TaskHelpers.cs

```diff
@@ -1,28 +1,65 @@
 ﻿using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Text;
+using System.Threading;
 using System.Threading.Tasks;
 using CRCTray.Communication;
 
 namespace CRCTray.Helpers
 {
     internal class TaskHelpers
     {
-        public static async Task<T> TryTask<T>(Func<T> function)
+        public static async Task<T> TryTask<T>(Func<T> function, bool retryable = false)

```

Comment:
I would not introduce this as a public function. Setting the `retryable` will not yield the expected results.

```cs
public static async Task<T> TryTask<T>(Func<T> function) {
{
    return await TryTask(function, false);
}

private static async Task<T> TryTask<T>(Func<T> function, bool retryable)
{
...
```